### PR TITLE
feat(cdk): Kotlin conversion cleanup

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
Cleanup after conversion of the Async framework to Kotlin:

* Use Logging logger
* Replace String.format with String interpolation
* Replace ImmutablePair with built-in Pair
* Replace list/map creation with built-in functions

### CODE REVIEW

**Comment by jdpgrailsdev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36449?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#36449** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36449?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#36447** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36447?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jdpgrailsdev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by jdpgrailsdev — COMMENTED** [review]
I'm not sure what is going on with the single test failure:

```
IntegrationRunnerTest > testNoInterruptOrphanThread() FAILED
    org.opentest4j.AssertionFailedError: expected: <1> but was: <9>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:531)
        at app//io.airbyte.cdk.integrations.base.IntegrationRunnerTest.testNoInterruptOrphanThread(IntegrationRunnerTest.java:414)
```

This test passes when I run the tests locally:

```
IntegrationRunnerTest > testNoInterruptOrphanThread() PASSED
````

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt [hunk 1]
```diff
@@ -43,79 +41,82 @@ import java.util.function.Consumer
...
-                        Field.of("name", JsonSchemaType.STRING),
+    companion object {
+        private const val RECORD_SIZE_20_BYTES = 20
+        private const val SCHEMA_NAME = "public"
+        private const val STREAM_NAME = "id_and_name"
+        private const val STREAM_NAME2 = STREAM_NAME + 2
+        private val STREAM1_DESC: StreamDescriptor =
+            StreamDescriptor().withNamespace(SCHEMA_NAME).withName(STREAM_NAME)
+
+        private val CATALOG: ConfiguredAirbyteCatalog =
```
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt:62]: Do we like non-const ALLCAPS names in Kotlin? Just curious, I don't care much.

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt [hunk 2]
```diff
@@ -336,16 +334,15 @@ class AsyncStreamConsumerTest {
...
 
     @Test
     internal fun deserializeAirbyteMessageWithBigDecimalAirbyteRecord() {
         val payload =
             Jsons.jsonNode(
-                java.util.Map.of(
-                    "foo",
-                    BigDecimal("1234567890.1234567890"),
+                mapOf(
+                    "foo" to BigDecimal("1234567890.1234567890"),
```
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt:345]: TIL! neat

**Review by postamar — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt:62]: Kotlin wants you to only use capitals for constants.  These were originally static fields in the Java code, hence why they are capitalized.  We could/should convert these to lower camel to be consistent with Kotlin styling, but that can be done later if we agree with their style.

Comment (postamar) [reply airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumerTest.kt:62]: Thanks for the explanation! I wasn't sure if it was just `const`s or not.

**Review by gisripa — APPROVED** [review]

**Comment by jdpgrailsdev** [comment]
### Merge activity

* **Mar 29, 9:58 AM EDT**: @jdpgrailsdev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/36449).

**Pull request merged**